crucible-llvm/src/Lang/Crucible/LLVM/SymIO.hs

```diff
@@ -61,6 +63,9 @@ module Lang.Crucible.LLVM.SymIO
   , callReadFileHandle
   , writeFileHandle
   , callWriteFileHandle
+  -- * File-related utilities
+  , allocateFileDescriptor
+  , lookupFileHandle
```

Comment:
I recommend augmenting the comment on `lookupFileHandle` to note that its continuation is called multiple times (because it is used in a symbolic branch). This may be surprising to callers if their continuation has side effects. I didn't note it before because users couldn't be surprised by an internal function.